app/views/layouts/_header.mobile.haml

```diff
@@ -4,32 +4,38 @@
       = link_to(image_tag("mobile/asterisk_white_mobile.png", class: "img-responsive"),
           stream_path, id: "header-title", class: "navbar-brand")
 
-      %ul.nav.navbar-nav#nav-badges
-        -# Notifications
-        %li
-          = link_to notifications_path, class: "badge-link", id: "notification-badge" do
-            %i.entypo-bell
-            - if current_user.unread_notifications.size > 0
-              %span.badge.badge-important#notification
-                = current_user.unread_notifications.size
+      - if user_signed_in?
+        %ul.nav.navbar-nav#nav-badges
+          -# Notifications
+          %li
+            = link_to notifications_path, class: "badge-link", id: "notification-badge" do
+              %i.entypo-bell
+              - if current_user.unread_notifications.size > 0
+                %span.badge.badge-important#notification
+                  = current_user.unread_notifications.size
 
-        -# Conversations
-        %li
-          = link_to conversations_path, class: "badge-link", id: "conversations-badge" do
-            %i.diaspora-custom-mail
-            - if current_user.unread_message_count > 0
-              %span.badge.badge-important#conversation
-                = current_user.unread_message_count
+          -# Conversations
+          %li
+            = link_to conversations_path, class: "badge-link", id: "conversations-badge" do
+              %i.diaspora-custom-mail
+              - if current_user.unread_message_count > 0
+                %span.badge.badge-important#conversation
+                  = current_user.unread_message_count
 
-        -# Publisher
-        %li
-          = link_to new_status_message_path, class: "badge-link", id: "compose-badge" do
-            %i.diaspora-custom-compose
+          -# Publisher
+          %li
+            = link_to new_status_message_path, class: "badge-link", id: "compose-badge" do
+              %i.diaspora-custom-compose
 
-        -# Menu
-        %li
-          %button.navbar-toggle#menu-badge{type: "button"}
-            %span.sr-only
-            %span.icon-bar
-            %span.icon-bar
-            %span.icon-bar
+          -# Menu
+          %li
+            %button.navbar-toggle#menu-badge{type: "button"}
+              %span.sr-only
+              %span.icon-bar
+              %span.icon-bar
+              %span.icon-bar
+
+      - else
+        - if AppConfig.settings.enable_registrations? && !current_page?(controller: "/registrations", action: :new)
+          %li= link_to t("devise.shared.links.sign_up"), new_user_registration_path, class: "login"
+        %li= link_to t("devise.shared.links.sign_in"), new_user_session_path, class: "login"
```

Comment:
You could use [these lines](https://github.com/diaspora/diaspora/blob/develop/app/views/layouts/_header.html.haml#L22-L25) from the desktop view. This would unify both versions and in `app/assets/stylesheets/mobile/header.scss` you will only have to keep `padding` and `font-weight` of the `.login` elements. All other CSS additions could be dropped. This would change the color of the header links but will unify them with the links from the desktop view.